.github/workflows/check-fuzz.yml

```diff
@@ -0,0 +1,42 @@
+# Smoke test to build fuzz targets.
+# Deserves its own action given that it depends on nightly
+# and there's currently no way to define multiple toolchains through the 
+# `rust-toolchain.toml` configuration file.
+name: Build Fuzz Targets
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  check-fuzz:
+    name: Build fuzz targets
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Read pinned nightly version
+        id: nightly_version
+        shell: bash
+        run: |
+          NIGHTLY_VERSION=$(cat pinned-nightly-version)
```

Comment:
What was the reason for putting this and the cargo fuzz version in a file instead of using a top-level `env` for both? I don't see them referenced outside this file.

Reply:
The reason is very subjective: I feel that's easier to update the version when it's defined at the root than having to modify the action itself.